projects/components/src/filtering/filter/parser/types/abstract-filter-parser.ts

```diff
@@ -0,0 +1,66 @@
+import { FilterAttribute } from '../../filter-attribute';
+import { FilterAttributeType } from '../../filter-attribute-type';
+import { FilterOperator } from '../../filter-operators';
+import { ParsedFilter } from '../parsed-filter';
+
+export abstract class AbstractFilterParser<TValue> {
+  public static supportedAttributeTypes: FilterAttributeType[];
+  public static supportedOperators: FilterOperator[];
+
+  protected abstract supportedAttributeTypes(): FilterAttributeType[];
```

Comment:
Seems strange to have this both as an instance method and static property.

Reply:
This is something I'm not thrilled about either. Its kind of confusing to the reader. The statics in the abstract class are never set. They are strictly there for the decorator to populate for the subclasses.

Reply:
I see in this change how they're used at an instance level - are they also being used at a static level? What's being done with the decorater-set values?

Reply:
You might be on to something here. This whole statics model with the builders (and now parsers) may not be needed anymore with the way I'm doing the lookups this time. I carried over the statics from the old code, but I don't think I'm going to need them.

Can we keep them for now, and once I've done the full end to end and determined I don't need them anymore I'll remove them?

Reply:
Take that back. They weren't even in the old code. I'm 99% sure I won't need them. I'll remove them next PR.

Reply:
Now that I'm looking at how I'm doing the lookup service, I won't actually need those. I'll remove them. Is it okay to do so in the next PR?